### PR TITLE
Convert esApply.Rnw to .Rmd

### PR DESCRIPTION
**Pull request by hsadia538**
Part of Sweave2rmd project.

- Creates a vignette in R markdown which replaces the sweave vignette.
- Attached are the HTML vignette and Sweave for comparison here
[esApply Rmd File](https://github.com/Bioconductor/Biobase/files/9748358/esApply.Introduction.html.zip)
[esApply Rnw File](https://bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/esApply.pdf)

### CODE REVIEW

**Comment by hsadia538** [comment]
Hi , Can you please take a look @Bioconductor/sweave2rmd

**Comment by hsadia538** [comment]
@jwokaty Can you take look, Thanks

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: Add a space after `golubEsets,`.

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: If no author is given, we don't have to include the author line. Remove line 3.

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: We use 

    date: `r format(Sys.Date(), "%B %d, %Y")` 

to have R write the date so we don't need to keep manually updating the date.

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: We should closely follow the code in the original document. We use `devtools` during development, but when a user is installing a package, the recommended way in Bioconductor is to use `BiocManager`. For the purposes of this vignette, we assume `Biobase` is installed and load it with `library`:

    ```{r message=FALSE, echo=FALSE}
    library(Biobase)
    data(sample.ExpressionSet)
    ```

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: I would group these into a single code block as in 

    ```{r}
    print(sample.ExpressionSet)
    print(exprs(sample.ExpressionSet)[1,])
    print(pData(sample.ExpressionSet)[1:2,1:3])
    ```

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: Improve the formatting by adding 4 spaces at the beginning of these lines.

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: I would also add spaces before line 45 to make it clear that it's still part of `rbind`.

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: Add a space after the comma in `1,medContr`

**Review by jwokaty — CHANGES_REQUESTED** [review]
Hi @hsadia538. Thanks for working on this vignette.

I noticed you accidentally committed some extra files, which should be removed. To avoid this I usually I usually explicitly `git add` specific files rather than doing `git add .`. We should only see the addition of the `.Rmd` file and the removal of the `.Rnw` file in this PR. To make that even clearer, they should be separate commits. Even if we think we should add a `.gitignore`, for example, pull requests should be specific to the feature that's being added, modified, or removed. 

Please make the changes that I noted separately. In addition please

- [x] Remove .Rbuildignore, .gitignore, Biobase.Rproj, file.tex
- [x] `git rm esApply.Rnw` and commit it separately with `git commit -m "Remove esApply.Rnw"`

When you're finished, push your changes to your remote branch and `@Bioconductor/Sweave2Rmd` so that we can do another review to check if it's ready to be merged or if further changes may be requested.

File: vignettes/esApply.Rmd
Comment (hsadia538) [reply vignettes/esApply.Rmd]: > We should closely follow the code in the original document. We use `devtools` during development, but when a user is installing a package, the recommended way in Bioconductor is to use `BiocManager`. For the purposes of this vignette, we assume it's installed and load it with `library`:
> 
> ```
> ```{r message=FALSE, echo=FALSE}
> library(Biobase)
> data(sample.ExpressionSet)
> ```
> ```

This command is run only with no output in the vignette. When I try to just add the code without the load command, the knit command does not work at all. Is there any workaround?

**Comment by hsadia538** [comment]
> Hi @hsadia538. Thanks for working on this vignette.
> 
> I noticed you accidentally committed some extra files, which should be removed. To avoid this I usually I usually explicitly `git add` specific files rather than doing `git add .`. We should only see the addition of the `.Rmd` file and the removal of the `.Rnw` file in this PR. To make that even clearer, they should be separate commits. Even if we think we should add a `.gitignore`, for example, pull requests should be specific to the feature that's being added, modified, or removed.
> 
> Please make the changes that I noted separately. In addition please
> 
> * [ ]  Remove .Rbuildignore, .gitignore, Biobase.Rproj, file.tex
> * [ ]  `git rm esApply.Rnw` and commit it separately with `git commit -m "Remove esApply.Rnw"`
> 
> When you're finished, push your changes to your remote branch and `@Bioconductor/Sweave2Rmd` so that we can do another review to check if it's ready to be merged or if further changes may be requested.

Thanks a lot for the detailed review. I will be making changes to this and other PR's as well and incorporate the feedback given.

File: vignettes/esApply.Rmd
Comment (jwokaty) [reply vignettes/esApply.Rmd]: You should make sure that you install `Biobase` first before trying to knit. In the RStudio console, you can do `devtools::install('Biobase')` to install Biobase then you should be able to knit. There should be no output from that code as we set `message=FALSE`.

Comment (hsadia538) [reply vignettes/esApply.Rmd]: What I meant is, when I remove the devtools::load_all("."). The knit fails until and unless I run this command before the code chunks.

**Comment by hsadia538** [comment]
@jwokaty  I have made the changes and updated the files. I am unable to tag `@Bioconductor/Sweave2Rmd` so tagging you.  Thanks

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: I didn't notice this before, but can you replace `Vignette Title` with `esApply Introduction`?

File: vignettes/esApply.Rmd
Comment (jwokaty) [reply vignettes/esApply.Rmd]: I can't see the image. Can you copy and paste it into a code block here? Or attach the image in the sweave2rmd slack?

**Comment by jwokaty** [comment]
> @jwokaty I have made the changes and updated the files. I am unable to tag `@Bioconductor/Sweave2Rmd` so tagging you. Thanks

I see. Thanks for letting me know about that.

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: This should be in two separate chunks as below because we don't want to echo or print the loading messages in the first chunk.

    ```{r message=FALSE, echo=FALSE}
    library(Biobase)
    data(sample.ExpressionSet)
    ```

    ```{r}
    print(sample.ExpressionSet)
    print(exprs(sample.ExpressionSet)[1,])
    print(pData(sample.ExpressionSet)[1:2,1:3])
    ```

File: vignettes/esApply.Rmd
Comment (jwokaty) [inline vignettes/esApply.Rmd]: This last code block should have `{r echo=FALSE}` to prevent the code from also being printed as we just want the output.

**Review by jwokaty — CHANGES_REQUESTED** [review]
Just a few more edits. I also want to ask you to reformat the authors in the `DESCRIPTION` file replacing the Authors line with the code below, which also adds you as a contributor. It should look like this

```
Authors@R: c(
    person("R.", "Gentleman", role="aut"),
    person("V.", "Carey", role = "aut"),
    person("M.", "Morgan", role="aut"),
    person("S.", "Falcon", role="aut"),
    person("Haleema", "Khan", role = "ctb"))
```

**Comment by hpages** [comment]
@jwokaty @hsadia538 We should probably keep the dot after the initials, so the render preserves the style of the current `Author` field:
```
> format(person("R", "Gentleman", role="aut"))
[1] "R Gentleman [aut]"
> format(person("R.", "Gentleman", role="aut"))
[1] "R. Gentleman [aut]"
```
Thanks

File: vignettes/esApply.Rmd
Comment (hsadia538) [reply vignettes/esApply.Rmd]: Here it is:
 ` Quitting from lines 23-27 (esApply.Rmd) 
Error in print(sample.ExpressionSet) : 
  object 'sample.ExpressionSet' not found
Calls: <Anonymous> ... withVisible -> eval_with_user_handlers -> eval -> eval -> print
Execution halted`

File: vignettes/esApply.Rmd
Comment (hsadia538) [reply vignettes/esApply.Rmd]: I have pushed the changes. please take a look. Thanks

File: vignettes/esApply.Rmd
Comment (jwokaty) [reply vignettes/esApply.Rmd]: I think this is because it needed `data(sample.ExpressionSet)`, which is how you load the data. Perhaps this was missing when you tried it?

File: vignettes/esApply.Rmd
Comment (jwokaty) [reply vignettes/esApply.Rmd]: @hsadia538 We need `{r message=FALSE, echo=FALSE}` for the first code block. After this, I believe it will be good to merge your pull request.

File: vignettes/esApply.Rmd
Comment (hsadia538) [reply vignettes/esApply.Rmd]: It was there. Are you able to run it locally? Maybe I am not installing it correctly😞

File: vignettes/esApply.Rmd
Comment (hsadia538) [reply vignettes/esApply.Rmd]: Updated and pushed changes!!

File: vignettes/esApply.Rmd
Comment (jwokaty) [reply vignettes/esApply.Rmd]: If you have time next week, we can do a slack huddle and you can walk me through what you're doing.

Comment (hsadia538) [reply vignettes/esApply.Rmd]: that'd be great!

**Review by jwokaty — CHANGES_REQUESTED** [review]
We're missing at the start of the second code block `print(sample.ExpressionSet)`. For reference, the two code chunks look like

    ```{r message=FALSE, echo=FALSE}
    library(Biobase)
    data(sample.ExpressionSet)
    ```

    ```{r}
    print(sample.ExpressionSet)
    print(exprs(sample.ExpressionSet)[1,])
    print(pData(sample.ExpressionSet)[1:2,1:3]) 
    ```

File: DESCRIPTION
Comment (mtmorgan) [inline DESCRIPTION:15]: sorry to make comments after this has been merged. I think the 'role' could be clarified with an 'comment' field, e.g.,
```
    person("Haleema", "Khan",
        role = "ctb"
        comment = "'esApply' vignette translation from Sweave to Rmarkdown / HTML"
    ))
```

File: vignettes/esApply.Rmd
Comment (mtmorgan) [inline vignettes/esApply.Rmd:3]: Here I think the format should be

```
author:
- name "R. Gentleman"
- name "V. Carey"
- name "M. Morgan"
- name "S. Falcon"
```
I think also that the translator deserves acknowledgement as an author with the role clearly specified. Adding the following would do this, although it is a 'hack' that re-uses the 'affiliation: ' tag (the affiliation appears as a footnote...)

```
- name: "H. Khan"
  affiliation: "Vignette translation from Sweave to Rmarkdown / HTML"
```

File: vignettes/esApply.Rmd
Comment (mtmorgan) [inline vignettes/esApply.Rmd:25]: I don't know whether it has consequences or not, but the blank line at the end of the code chunk should be avoided.

File: vignettes/esApply.Rmd
Comment (mtmorgan) [inline vignettes/esApply.Rmd:29]: These `print()` statements (and in other code chunks) are not necessary and should be removed.

File: DESCRIPTION
Comment (mtmorgan) [inline DESCRIPTION:16]: The `Authors@R` format should include the maintainer (`role = "cre"`). I think the following works and should be added to the `Authors@R` entry

```
    person(
        "Bioconductor Package Maintainer",
        role = "cre",
        email = "maintainer@bioconductor.org"
    )
```
The line `Maintainer: Bioconductor Package Maintainer` and email on the following line should be removed.

Comment (hsadia538) [reply DESCRIPTION:16]: Hello, I see that it is merged now. What should I do. making the changes as per the latest review, will it affect anything?

Comment (jwokaty) [reply DESCRIPTION:16]: @hsadia538  Could I ask you to make another pull request with these changes? @mtmorgan Thanks for your input, which helps me a lot, and your advocacy to make sure contributors get credit.

Comment (hsadia538) [reply DESCRIPTION:16]: Sure thanks for clarifying. I will make a new one.

**Comment by LiNk-NY** [comment]
FWIW, see the errors here based on changes to the `DESCRIPTION` file (noted by Martin above)
https://master.bioconductor.org/checkResults/3.16/bioc-LATEST/Biobase/nebbiolo2-install.html

**Comment by jwokaty** [comment]
@LiNk-NY Thank you. This should be resolved when we close #17

**Pull request merged**
